### PR TITLE
feat(query): "Cleartext Credentials with Basic Authentication" for OpenAPI

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
**Proposed Changes**
- Add "Cleartext Credentials With Basic Authentication for OpenAPI"
- Checks if servers URL uses HTTPS
- Checks if securityScheme is set as basic authentication
- Checks if global security is empty

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/cleartext_credentials_with_basic_auth/query.rego [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+CxPolicy[result] {
+	doc := input.document[i]
+	openapi_lib.check_openapi(doc) != "undefined"
+
+	openapi_lib.is_basic_authentication(doc.components.securitySchemes)
+
+	server := doc.servers[j]
+	openapi_lib.is_unsecure_http(server.url)
+
+	sec := doc.security[k][schemeName]
```
Comment (felipe-avelar) [inline assets/queries/openAPI/cleartext_credentials_with_basic_auth/query.rego:14]: I think we should check if security is an array or an object

File: assets/queries/openAPI/cleartext_credentials_with_basic_auth/query.rego [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	is_array(sec)
+	sec == []
```
Comment (felipe-avelar) [inline assets/queries/openAPI/cleartext_credentials_with_basic_auth/query.rego:16]: I think we should check if `sec == {}` too...

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

**Comment by rogeriopeixotocx** [comment]
Closed in favor of https://github.com/Checkmarx/kics/pull/3349

**Pull request closed without merging**